common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -8026,7 +8062,9 @@ static jlong NativeCrypto_SSL_set_timeout(JNIEnv* env, jclass, jlong ssl_address
     }
 
     // Convert to seconds
-    long timeout = millis / 1000;
+    static const jlong INT_MAX_AS_JLONG = static_cast<jlong>(INT_MAX);
+    uint32_t timeout = static_cast<uint32_t>(std::max(0,
```

Comment:
I think windows.h also has min/max macros that mess things up. We build BoringSSL with `NOMINMAX` defined next to `WIN32_LEAN_AND_MEAN`. That might be needed too?